WebApplication/Services/AdoptProjectService.cs

```diff
@@ -12,45 +15,72 @@ public class AdoptProjectService
     {
         private readonly ILogger<AdoptProjectService> _logger;
         private readonly ProjectService _projectService;
+        private readonly ProjectWork _projectWork;
+        private readonly Uploads _uploads;
+        private readonly UserResolver _userResolver;
+        private readonly DtoGenerator _dtoGenerator;
 
-        public AdoptProjectService(ILogger<AdoptProjectService> logger, ProjectService projectService)
+        public AdoptProjectService(ILogger<AdoptProjectService> logger, ProjectService projectService, 
+            Uploads uploads, ProjectWork projectWork, UserResolver userResolver, DtoGenerator dtoGenerator)
         {
             _logger = logger;
             _projectService = projectService;
+            _uploads = uploads;
+            _projectWork = projectWork;
+            _userResolver = userResolver;
+            _dtoGenerator = dtoGenerator;
         }
 
         /// <summary>
         /// https://jira.autodesk.com/browse/INVGEN-45256
         /// </summary>
         /// <param name="payload">project configuration with parameters</param>
-        public string AdoptProjectWithParameters(AdoptProjectWithParametersPayload payload)
+        public async Task<string> AdoptProjectWithParametersAsync(AdoptProjectWithParametersPayload payload)
         {
-            _logger.LogInformation($"adopting project {payload.Name}");
+            //TODO: check whether project already exists. If yes, skip upload and adoption and continue with parameters update.
+            var packageId = CreateProjectAsync(payload).Result;
+
+            (ProjectInfo projectInfo, string fileName) = _uploads.GetUploadData(packageId);
+            _uploads.ClearUploadData(packageId);
+
+            var id = Guid.NewGuid().ToString();
+            await _projectService.AdoptProject(projectInfo, fileName, id);
+
+            await _projectWork.DoSmartUpdateAsync(payload.Config, payload.Name);
+
+            return packageId;
+        }
+
+        /// <summary>
+        /// Kind of a facade for ProjectService.CreateProject
+        /// </summary>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
+        private async Task<string> CreateProjectAsync(DefaultProjectConfiguration configuration)
+        {
+            _logger.LogInformation($"adopting project {configuration.Name}");
 
             using var localFile = new TempFile();
             var localFileName = localFile.Name;
             using (var client = new WebClient())
```

Comment:
WebClient is obsolete for netcore. Please use `HttpClient` and `HttpClientFactory` (and there is a [helper method](https://github.com/Developer-Autodesk/forge-configurator-inventor/blob/05b415b0a619ce84b864bb3688df37f83702771b/WebApplication/Utilities/Web.cs#L33) you can use)

Reply:
I'll change it.